vmtest/vm.py

```diff
@@ -54,50 +51,59 @@ def force_stop(self):
         Stop the VM and clean up any resources that were created when setting up and starting the machine.
         """
 
-    def _get_ssh_transport(self, user, password="", keyfile=None):
+    def _sshpass(self, password):
+        if not password:
+            return []
+        return ["sshpass", "-p", password]
+
+    def _ensure_ssh(self, cmd, user, password="", keyfile=None):
         if not self.running():
             self.start()
-        client = SSHClient()
-        client.set_missing_host_key_policy(AutoAddPolicy)
-        # workaround, see https://github.com/paramiko/paramiko/issues/2048
-        pkey = None
-        if keyfile:
-            for klass in paramiko.key_classes:
-                try:
-                    pkey = klass.from_private_key_file(keyfile)
-                    break
-                except paramiko.ssh_exception.SSHException:
-                    continue
-            if pkey is None:
-                raise RuntimeError(f"cannot load {keyfile}, tried {paramiko.key_classes}")
-        client.connect(
-            self._address, self._ssh_port,
-            user, password, pkey=pkey,
-            allow_agent=False, look_for_keys=False)
-        return client.get_transport()
+        for i in range(3):
+            try:
+                ret, _ = self._run("true", user, password, keyfile)
+                if ret == 0:
+                    return
+            except Exception as e:
+                print(f"ssh not ready {e}")
+                time.sleep(1)
```

Comment:
I'd wait a bit longer than 1s, we're using 10s in image so maybe just make it consistent everywhere?